### PR TITLE
feat: Updated Disclaimer screen with trial option

### PR DESCRIPTION
**Pull request by rijuma**
Ticket: [COSMO-570](https://2u-internal.atlassian.net/browse/COSMO-570)🔒

# Description
We are going to introduce a limited time free trial for audit users. This PR prepares the UI for that.
I've added the trial message to be shown by a prop, but it's currently disabled. This new UI element with the CTA will be tackled in ticket [COSMO-602](https://2u-internal.atlassian.net/browse/COSMO-602)🔒 but I just took the opportunity to include it since it was just removing part of the mockup.

I took the opportunity to add some extra tweaks to the general UI to look closer to the expected [figma design](https://www.figma.com/design/5D53Q5ypmwp1Rudr1ruw4i/Xpert-Learning-Assistant?node-id=44-254&t=ru3gy0sFycrWP0yz-1).

# Screenshots

## Without trial message
image

## With trial message
image

### CODE REVIEW

**Comment by alangsto** [comment]
I'll take a closer look at this tomorrow, but the message input should be closer to the bottom of the chat sidebar! It looks like it's position moved (and this also differs from the Figma mocks) in the screenshots you provided.

File: src/components/Disclosure/Disclosure.test.jsx [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+const PRIVACY_POLICY_URL = 'https://some.url/policy';
+jest.mock('@edx/frontend-platform/config', () => ({
+  ensureConfig: jest.fn(),
+  getConfig: () => ({ PRIVACY_POLICY_URL }),
+}));
+
+describe('<TrialDisclosure />', () => {
+  let container;
+
+  describe('When trial upgrade is not being showed', () => {
```
Comment (alangsto) [inline src/components/Disclosure/Disclosure.test.jsx:16]:
```suggestion
  describe('When trial upgrade is not being shown', () => {
```

File: src/components/Disclosure/Disclosure.test.jsx [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+      expect(upgrade).not.toBeInTheDocument();
+    });
+
+    it('should match snapshot', () => {
+      expect(container).toMatchSnapshot();
+    });
+  });
+
+  describe('When trial upgrade being showed', () => {
```
Comment (alangsto) [inline src/components/Disclosure/Disclosure.test.jsx:39]:
```suggestion
  describe('When trial upgrade being shown', () => {
```

File: src/components/Disclosure/index.jsx [hunk 3]
```diff
@@ -1,42 +1,53 @@
...
+import { QuestionAnswerOutline, LightbulbCircle, AutoAwesome } from '@openedx/paragon/icons';
 import { ensureConfig, getConfig } from '@edx/frontend-platform/config';
 
 import './Disclosure.scss';
 
 ensureConfig(['PRIVACY_POLICY_URL']);
 
-const Disclosure = ({ children }) => (
-  <div className="disclosure d-flex flex-column align-items-stretch px-4 py-3">
+const Disclosure = ({ children, showTrial }) => (
```
Comment (alangsto) [inline src/components/Disclosure/index.jsx:12]: Should this prop be passed in from the Sidebar component? No worries if this logic is included as part of your CTA work.

**Comment by ilee2u** [comment]
FYI I'm going to be rebasing my PR (https://github.com/edx/frontend-lib-learning-assistant/pull/72) onto this one so our changes are synced.

**Review by alangsto — COMMENTED** [review]
A few nits, but otherwise looks good! Would you mind updating the screenshots included in the PR with the updated UX? Thank you!

File: src/components/Disclosure/index.jsx (hunk 3 above)
Comment (rijuma) [reply src/components/Disclosure/index.jsx]: It's s not. the prop is outdated and it was fixed in the other PR pointing to this one (https://github.com/edx/frontend-lib-learning-assistant/pull/74). I've removed it. It's fixed now. Sorry.

**Comment by rijuma** [comment]
> A few nits, but otherwise looks good! Would you mind updating the screenshots included in the PR with the updated UX? Thank you!

Updated!

File: src/components/Disclosure/index.jsx [hunk 4]
```diff
@@ -1,53 +1,85 @@
...
-        <Icon src={Chat} className="m-2" />
-        <div>
-          Stuck on a concept? Need more clarification on a complicated topic?
-          <br />
-          Ask Xpert a question!
+const Disclosure = ({ children }) => {
+  const { upgradeable, upgradeUrl, auditTrialLengthDays } = useCourseUpgrade();
+  const { track } = useTrackEvent();
+
+  const handleClick = () => track('edx.ui.lms.learning_assistant.message');
```
Comment (alangsto) [inline src/components/Disclosure/index.jsx:17]: Just noticed this as I'm working through my rebase/refactor, but I don't think this is the correct event name. The `message` suffix would make more sense for segment events tracked for each message sent, but something like `upgrade_click` might be more suitable for this specific event.

File: src/hooks/use-course-upgrade.js [hunk 5]
```diff
@@ -0,0 +1,40 @@
+import { useContext } from 'react';
+import { useModel } from '@src/generic/model-store'; // eslint-disable-line import/no-unresolved
+import { useSelector } from 'react-redux';
+import { CourseInfoContext } from '../context';
+
+const millisecondsInOneDay = 24 * 60 * 60 * 1000; // hours*minutes*seconds*milliseconds
+
+export default function useCourseUpgrade() {
```
Comment (schenedx) [inline src/hooks/use-course-upgrade.js:8]: Could you add comments on the key utility of this hook?

File: src/components/Disclosure/index.jsx (hunk 4 above)
Comment (rijuma) [reply src/components/Disclosure/index.jsx]: Thanks for noticing this. I've changed it for `edx.ui.lms.learning_assistant.disclosure_upgrade_click`. My thoughts on this is to be able to tell whether a student updated straight from the disclosure message or from the other instances (Upgrade through the banner or the paywall when the trial expired).

File: src/hooks/use-course-upgrade.js (hunk 5 above)
Comment (rijuma) [reply src/hooks/use-course-upgrade.js:31]: You're right. Thanks for the recommendation. I've added the `JSDoc` comments for both new hooks.

**Review by ilee2u — APPROVED** [review]
Tested this with my PR. Everything works great, so LGTM 🚀

**Review by schenedx — APPROVED** [review]
👍

**Pull request merged**